### PR TITLE
Update CLIPFeatureExtractor to convert PIL image to RGB

### PR DESCRIPTION
**Pull request by hengkuanwee**
# What does this PR do?

Currently PIL images with `RGBA` format throws an error when being processed by CLIPFeatureExtractor. 

CLIPFeatureExtractor.normalize() throws the following error:
`.../sentence-transformer/lib/python3.9/site-packages/transformers/image_utils.py", line 185, in normalize
    return (image - mean) / std
ValueError: operands could not be broadcast together with shapes (4,224,224) (3,) `

The original [clip model preprocesses PIL Images](https://github.com/openai/CLIP/blob/main/clip/clip.py#L74) by converting all PIL images into `RGB` format.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/clip/test_feature_extraction_clip.py [hunk 1]
```diff
@@ -227,3 +231,64 @@ def test_call_pytorch(self):
...
+        self.expected_encoded_image_num_channels = 3
+
+    @property
+    def feat_extract_dict(self):
+        return self.feature_extract_tester.prepare_feat_extract_dict()
+
+    def test_feat_extract_properties(self):
+        feature_extractor = self.feature_extraction_class(**self.feat_extract_dict)
+        self.assertTrue(hasattr(feature_extractor, "do_resize"))
+        self.assertTrue(hasattr(feature_extractor, "size"))
```
Comment (patil-suraj) [inline tests/clip/test_feature_extraction_clip.py:253]: Nice tests!

**Review by patil-suraj — APPROVED** [review]
LGTM, thank you for adding this!

@sgugger @NielsRogge should we add this to all vision feature extractors ?

File: src/transformers/models/clip/feature_extraction_clip.py [hunk 2]
```diff
@@ -142,11 +146,13 @@ def __call__(
...
             images = [images]
 
-        # transformations (resizing + center cropping + normalization)
+        # transformations (resizing + center cropping + convert rgb + normalization)
         if self.do_resize and self.size is not None and self.resample is not None:
             images = [self.resize(image=image, size=self.size, resample=self.resample) for image in images]
         if self.do_center_crop and self.crop_size is not None:
             images = [self.center_crop(image, self.crop_size) for image in images]
+        if self.do_convert_rgb:
+            images = [self.convert_rgb(image) for image in images]
```
Comment (sgugger) [inline src/transformers/models/clip/feature_extraction_clip.py:155]: This should be at the very beginning of the pipeline I think, even before the resize.

File: src/transformers/models/clip/feature_extraction_clip.py [hunk 3]
```diff
@@ -156,6 +162,20 @@ def __call__(
...
+
+        Args:
+            image (`PIL.Image.Image` or `np.ndarray` or `torch.Tensor`):
+                The image to convert.
+        """
+        self._ensure_format_supported(image)
+        if not isinstance(image, Image.Image):
+            image = self.to_pil_image(image)
+
+        return image.convert("RGB")
```
Comment (sgugger) [inline src/transformers/models/clip/feature_extraction_clip.py:177]: I don't think we should force a conversion for non-PIL.Image types. If the user has passed a NumPy array or a torch Tensor, they will have converted it to the format of their choice already.

**Review by sgugger — COMMENTED** [review]
Thanks for your PR!
I don't think we should force a conversion to PIL Images for this (definitely all for applying it on PIL Images however!).

File: src/transformers/models/clip/feature_extraction_clip.py (hunk 3 above)
Comment (sgugger) [reply src/transformers/models/clip/feature_extraction_clip.py:177]: Yup, basically changing
```
        if not isinstance(image, Image.Image):
            image = self.to_pil_image(image)
```
to
```
        if not isinstance(image, Image.Image):
            return image
```

**Comment by hengkuanwee** [comment]
@patil-suraj @sgugger any idea why the `Build dev documentation / build_and_package` check failed? based on the error log, I initially suspected that master branch is out of sync (i.e. there are new commits), however, I just verified that it is up to date 🤔

**Comment by patil-suraj** [comment]
You can rebase to master to solve the failing test.

**Comment by hengkuanwee** [comment]
I checkout my branch and ran `git rebase master`, it returns `Current branch convert-pil-image-to-rgb-clip-model is up to date.`

**Comment by patil-suraj** [comment]
You should rebase with `huggingface/transformers`, for example if your remote is called `upstream` you can run the following commands to rebase.

```bash
git fetch upstream
git rebase upstream/master
``` 

and then push.

**Comment by patil-suraj** [comment]
Hey @hengkuanwee , this slipped through the cracks. Do you maybe want to open new PR for this ?

**Comment by hengkuanwee** [comment]
hi @patil-suraj, sure i'll open a new PR for this!

**Pull request closed without merging**